specs/Theme.md

```diff
@@ -0,0 +1,130 @@
+Theme API
+===
+
+# Background
+This API's main use is to set the overall theme for WebView2's which are associated with a profile. The options are similar to Edge: match the system default theme, change to light theme, or change to dark theme.
+This sets the theme for WebView2 UI like dialogs, prompts, and context menus by setting the ['prefers-color-scheme'](https://developer.mozilla.org/en-US/docs/Web/CSS/@media/prefers-color-scheme) [media feature](https://developer.mozilla.org/en-US/docs/Web/CSS/Media_Queries/Using_media_queries#media_features). Websites typically query for this setting in order to set CSS themes for light/dark.
+
+Please note this API will only set the overall appearance, but not theme as defined in the Edge browser.
+For reference, in the screenshot below, this API is meant to expose the Overall Appearance section as a WebView2 API.
+
+![Edge Settings Appearance Page](https://github.com/MicrosoftEdge/WebView2Feedback/blob/api-appearance/specs/images/EdgeSettingsAppearance.png)
+
+# Description
+We propose extending `CoreWebView2Profile` to include a `Theme` property. This property will apply
+to any WebView2's which are associated with this profile.
+
+# Examples
+
+## C++
+
+```cpp
+wil::com_ptr<ICoreWebView2> m_webView;
+
+void ViewComponent::SetTheme(COREWEBVIEW2_THEME_KIND value)
+{
+    wil::com_ptr<ICoreWebView2_7> webView7;
+    webView7 = m_webView.try_query<ICoreWebView2_7>();
+
+    if (webView7)
+    {
+        wil::com_ptr<ICoreWebView2Profile> profile;
+        CHECK_FAILURE(webView7->get_Profile(&profile));
+
+        auto profile3 = profile.try_query<ICoreWebView2Profile3>();
+        if (profile3)
+        {
+            profile3->put_Theme(value);
+        }
+    }
+}
+```
+
+## C#
+
+```c#
+
+private CoreWebView2 m_webView;
+
+void SetTheme(CoreWebView2ThemeKind value)
+{
+    // Check for runtime support
+    try
+    {
+        m_webView.Profile.Theme = value;
+    }
+    catch (NotImplementedException exception)
```

Comment:
Does this work? According to the [C# documentation for HRESULT-to-Exception mapping](https://docs.microsoft.com/en-us/dotnet/framework/interop/how-to-map-hresults-and-exceptions?redirectedfrom=MSDN), `E_NOINTERFACE` maps to `InvalidCastException`, not `NotImplementedException`.

Reply:
I don't think it makes sense for the public samples to try and be version-adaptive like this. We shouldn't be demonstrating that apps have exceptions in non-exceptional conditions.

Reply:
You are right, it should be `InvalidCastException`. I think Mike makes a good point about not being version adaptive in our samples. This is something we do pretty frequently in our sample apps however, @david-risney what are your thoughts on this?

Reply:
We can move the exception checks up the stack in the sample app code so it doesn't show up in the snippets if we don't want to demonstrate that. Mike's feedback makes sense. We should have one place in our docs to talk about version-adaptive checks rather than putting them in all C# sample code.